packages/offix-datastore/src/DataStore.ts

```diff
@@ -1,74 +1,77 @@
 import { Model, ModelConfig } from "./Model";
-import { LocalStorage } from "./storage";
-import { createGraphQLClient, GraphQLClientConfig } from "./replication/client/GraphQLClient";
-import { buildGraphQLCRUDQueries } from "./replication/graphqlcrud/buildGraphQLCRUDQueries";
+import { LocalStorage, StorageAdapter } from "./storage";
+import { createGraphQLClient } from "./replication/GraphQLClient";
+import { buildGraphQLCRUDQueries } from "./replication/documents/buildGraphQLCRUDQueries";
 import { IReplicator } from "./replication/api/Replicator";
-import { MutationReplicationEngine } from "./replication";
-import { GraphQLCRUDReplicator } from "./replication/graphqlcrud/GraphQLCRUDReplicator";
+import { GraphQLCRUDReplicator } from "./replication/GraphQLCRUDReplicator";
 import { IndexedDBStorageAdapter } from "./storage/adapters/IndexedDBStorageAdapter";
+import { DataStoreConfig } from "./DataStoreConfig";
+import { createLogger, enableLogger } from "./utils/logger";
+
+const logger = createLogger("DataStore");
+// TODO disable logging before release
+enableLogger();
+
+const metadataStoreName = "model_metadata"
 
 /**
- * Configuration Options for DataStore
+ * Custom implementation
  */
-export interface DataStoreConfig {
-  /**
-   * The Database name
-   */
-  dbName: string;
-
+export interface CustomEngines {
```

Comment:
Customization options for engines. Exposed that so we can support different replication and store. I'm simplifying things a lot. 

If you want to have different replication/client etc. you will need to swap entire replication (or extend ours) rather than building some swappable abstraction within replication itself.

Reply:
this is no longer valid. We need to wrap this later into different queries